### ISSUE TITLE
Fail - Unable to verify

### ISSUE THREAD

**Issue by olegabr** [rkalis/truffle-plugin-verify#21]
Token is compiled and deployed.
When I try to verify it I get:

```
truffle run verify TestToken --network mainnet
> Warning: possible unsupported (undocumented in help) command line option: --network
Verifying TestToken
Fail - Unable to verify

Failed to verify 1 contract(s): TestToken
```

contracts/Token.sol
==============

```sol
pragma solidity ^0.5.1;

import "openzeppelin-solidity/contracts/token/ERC777/ERC777.sol";

contract TestToken is ERC777(
  "Test token",
  "TSX",
  new address[](0)
) {
  constructor() public {
    _mint(
      msg.sender,
      0x476Bb28Bc6D0e9De04dB5E19912C392F9a76535d,
      1000000000 * 10 ** uint256(decimals()),
      "",
      ""
    );
  }
}
```

package.json
==========

```json
{
  "name": "ERC777",
  "version": "1.0.0",
  "description": "ERC777 token",
  "main": "truffle-config.js",
  "directories": {
    "test": "test"
  },
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "olegabr@ethereumico.io",
  "license": "ISC",
  "dependencies": {
    "@truffle/hdwallet-provider": "1.0.18",
    "openzeppelin-solidity": "2.3.0",
    "truffle-plugin-verify": "0.3.3"
  },
  "devDependencies": {
    "chai": "^4.2.0",
    "chai-bn": "^0.1.1"
  }
}
```

migrations/2_deploy_contracts.js
=========================

```javascript
const TestToken = artifacts.require("TestToken");

module.exports = function(deployer, network, accounts) {
  deployer.deploy(TestToken).then(function() {
      const aTestTokenAddress = TestToken.address;
      console.log("Token address:", aTestTokenAddress)
      return TestToken.deployed();
  });
};
```

truffle-config.js
===========

```javascript

const HDWalletProvider = require('@truffle/hdwallet-provider');
const infuraKey = "YOUR_INFURA_KEY_HERE"; // TODO: place your infura.io API Key here

const fs = require('fs');
// TODO: place your private key or seed phrase in a file named .secret in the directory root
const mnemonic = fs.readFileSync(".secret").toString().trim();
// const mnemonic = "";

module.exports = {
  plugins: [
    'truffle-plugin-verify'
  ],
  api_keys: {
    etherscan: 'YOUR_ETHERSCAN_KEY_HERE' // TODO: place your etherscan.io API key here
  },
  /**
   * Networks define how you connect to your ethereum client and let you set the
   * defaults web3 uses to send transactions. If you don't specify one truffle
   * will spin up a development blockchain for you on port 9545 when you
   * run `develop` or `test`. You can ask a truffle command to use a specific
   * network from the command line, e.g
   *
   * $ truffle test --network <network-name>
   */

  networks: {
    // Useful for testing. The `development` name is special - truffle uses it by default
    // if it's defined here and no other network is specified at the command line.
    // You should run a client (like ganache-cli, geth or parity) in a separate terminal
    // tab if you use this network and you must also set the `host`, `port` and `network_id`
    // options below to some value.
    //
    development: {
     host: "127.0.0.1",     // Localhost (default: none)
     // port: 8545,            // Standard Ethereum port (default: none)
     port: 9545,            // Standard Ethereum port (default: none)
     network_id: "*",       // Any network (default: none)
     gas: 6721975,           // Gas sent with each transaction (default: ~6700000)
     gasPrice: 1,  // wei
     skipDryRun: false,
    },

    // Another network with more advanced options...
    // advanced: {
      // port: 8777,             // Custom port
      // network_id: 1342,       // Custom network
      // gas: 8500000,           // Gas sent with each transaction (default: ~6700000)
      // gasPrice: 20000000000,  // 20 gwei (in wei) (default: 100 gwei)
      // from: <address>,        // Account to send txs from (default: accounts[0])
      // websockets: true        // Enable EventEmitter interface for web3 (default: false)
    // },

    // Useful for deploying to a public network.
    // NB: It's important to wrap the provider as a function.
    ropsten: {
      provider: () => new HDWalletProvider(mnemonic, "https://ropsten.infura.io/v3/" + infuraKey),
      network_id: 3,       // Ropsten's id
      gas: 5500000,        // Ropsten has a lower block limit than mainnet
      gasPrice: 20 * 1e9,  // 20 gwei (in wei) (default: 100 gwei)
      confirmations: 1,    // # of confs to wait between deployments. (default: 0)
      timeoutBlocks: 200,  // # of blocks before a deployment times out  (minimum/default: 50)
      skipDryRun: true     // Skip dry run before migrations? (default: false for public nets )
    },

    rinkeby: {
      provider: () => new HDWalletProvider(mnemonic, "https://rinkeby.infura.io/v3/" + infuraKey),
      network_id: 4,       // Ropsten's id
      gas: 5500000,        // Ropsten has a lower block limit than mainnet
      gasPrice: 20 * 1e9,  // 20 gwei (in wei) (default: 100 gwei)
      confirmations: 1,    // # of confs to wait between deployments. (default: 0)
      timeoutBlocks: 200,  // # of blocks before a deployment times out  (minimum/default: 50)
      skipDryRun: true     // Skip dry run before migrations? (default: false for public nets )
    },

    // Useful for deploying to a public network.
    // NB: It's important to wrap the provider as a function.
    mainnet: {
      provider: () => new HDWalletProvider(mnemonic, "https://mainnet.infura.io/v3/" + infuraKey),
      network_id: 1,       // mainnet's id
      gas: 8500000,        // Ropsten has a lower block limit than mainnet
      gasPrice: 2 * 1e9,  // 11 gwei (in wei) (default: 100 gwei)
      confirmations: 1,    // # of confs to wait between deployments. (default: 0)
      timeoutBlocks: 2000,  // # of blocks before a deployment times out  (minimum/default: 50)
      skipDryRun: true     // Skip dry run before migrations? (default: false for public nets )
    },

    // Useful for private networks
    // private: {
    //   provider: () => new HDWalletProvider(mnemonic, `http://127.0.0.1:8545`),
    //   network_id: 6660001,   // This network is yours, in the cloud.
    //   production: true    // Treats this network as if it was a public net. (default: false)
    // }
  },

  // Set default mocha options here, use special reporters etc.
  mocha: {
    // timeout: 100000
  },

  // Configure your compilers
  compilers: {
    solc: {
      version: "0.5.2",    // Fetch exact version from solc-bin (default: truffle's version)
      // docker: true,        // Use "0.5.1" you've installed locally with docker (default: false)
      settings: {          // See the solidity docs for advice about optimization and evmVersion
       optimizer: {
         enabled: true,
         runs: 400
       },
       // evmVersion: "byzantium",
      }
    }
  }
}
```

**Comment by rkalis** [comment]
Thanks for the report! Quick few questions: does it work on testnet (e.g. rinkeby), and does it make a difference if you specifically specify `evmVersion: 'byzantium'`?

**Comment by olegabr** [comment]
It failed for ropsten too. But I'm not sure it takes the network into account in the first place. Note the Warning about the --network key.

I have not tried the `byzantium` key.

I've succeeded (https://etherscan.io/address/0xf886A23D823b37ba338b4809D9eE1D94c1478089#code) with the https://github.com/RyuuGan/sol-merger tool, but with this workaround: https://github.com/RyuuGan/sol-merger/issues/21

**Comment by rkalis** [comment]
Hmm if you needed to manually add something into the output of sol-merger, then I think that is the issue with truffle-plugin-verify as well, since the plugin uses sol-merger under the hood. So I'll wait to see if that can be solved on the sol-merger side, otherwise I'll research further.

**Comment by rkalis** [comment]
About the unsupported flag warning, that was a Truffle bug that was fixed in [Truffle@5.0.38](https://github.com/trufflesuite/truffle/releases/tag/v5.0.38). The warning has no impact on the actual functioning of the plugin though, so the `--network` parameter is used correctly.

**Comment by olegabr** [comment]
The https://github.com/RyuuGan/sol-merger/issues/21 issue is fixed.
